src/Microsoft.PowerShell.Commands.Utility/commands/utility/UtilityCommon.cs

```diff
@@ -206,6 +206,58 @@ private set
         /// </summary>
         public string Path { get; private set; }
 
+        /// <summary>
+        /// Gets the hexadecimal representation of the <see cref="Offset64"/> value.
+        /// </summary>
+        public string HexOffset { get => string.Format("{0:X16}", Offset64); }
+
+        private const int BytesPerLine = 16;
+
+        /// <summary>
+        /// Gets a space-delimited string of the <see cref="Bytes"/> in this <see cref="ByteCollection"/>
+        /// in hexadecimal format.
+        /// </summary>
+        public string HexBytes
+        {
+            get
+            {
+                StringBuilder line = new StringBuilder(BytesPerLine * 3);
+
+                foreach (var currentByte in Bytes)
+                {
+                    line.AppendFormat("{0:X2} ", currentByte);
+                }
+
+                return line.ToString().Trim();
+            }
+        }
+
+        /// <summary>
+        /// Gets the ASCII string representation of the <see cref="Bytes"/> in this <see cref="ByteCollection"/>.
+        /// </summary>
+        /// <value></value>
+        public string AsciiBytes
+        {
+            get
+            {
+                StringBuilder ascii = new StringBuilder(BytesPerLine);
+
+                foreach (var currentByte in Bytes)
+                {
+                    if ((currentByte >= 0x20) && (currentByte <= 0xFE))
+                    {
+                        ascii.Append((char)currentByte);
+                    }
+                    else
+                    {
+                        ascii.Append('.');
```

Comment:
Why not replacement character U+FFFD?

Reply:
Just following the existing behaviour, really. Would that be preferred? That's the boxed question-mark character, right?

:thinking: does `char` go that high? Never really tried :D

Reply:
It is Unicode standard "Replacement character", not a question mark.
We had a discussion with @SteveL-MSFT in the issue #7777.

Reply:
https://github.com/PowerShell/PowerShell/issues/7777#issuecomment-497195500

>So right fix is (1) convert byte to Unicode char, (2) check char.IsControl, (3) replace controls with U+FFFD

So that seems simple enough. There's almost mention of replacing `0x00` with a space instead, is this also something we want to do?

Reply:
This is gonna mean I have to edit tests too.. I'll have to fix those this evening, but got this sorted at least I think. 😄